### ISSUE TITLE
Safari 8 unable to load some packages.

### ISSUE THREAD

**Issue by jdalton** [jspm/jspm-cli#551]
Visit http://jsbin.com/cobevi/1/ and open the web console in Safari 8.

You'll see it errors loading the `difference` module.
![image](https://cloud.githubusercontent.com/assets/4303/6423133/7d632cda-be96-11e4-8f65-5073424c5c44.png)

This does not happen in other browsers.

```html
<script src="//jspm.io/system@0.9.js"></script>
<script>
System.import('npm:lodash-fp').then(function(fp) {
  _ = fp;
});
</script>
```

**Comment by jdalton** [comment]
![image](https://cloud.githubusercontent.com/assets/4303/6423133/7d632cda-be96-11e4-8f65-5073424c5c44.png)

**Comment by guybedford** [comment]
Wow, this is really strange. Note that the latest SystemJS is 0.14, but it still happens with that.

So if I do:

```javascript
System.import('npm:lodash-fp/mixin').then(function() {
  System.import('npm:lodash-fp').then(console.log.bind(console));
});
```

It works fine. Similarly if I load `npm:lodash-compat@3` first it also works fine.

So the problem is in neither dependency, but rather in having both...

Next I tried writing a local file containing - 

```javascript
/* */
"format cjs";module.exports=require("npm:lodash-fp/mixin")(require("npm:lodash-compat@^3.0.0").runInContext());
//# sourceMappingURL=index.js.map
```

This worked absolutely fine to load.

So somehow the difference between the above, and:

```javascript
/* */
"format cjs";module.exports=require("./mixin")(require("npm:lodash-compat@^3.0.0").runInContext());
//# sourceMappingURL=index.js.map
```

In the repo itself is enough to cause the issue reliably!

I also checked the dependency analysis (by inspecting `System.failed`) and it is definitely determining and resolving the dependencies correctly (`lodash-fp/index` having the correct dependencies and normalized dependency names).

The next debugging step would be to clone the npm repo into a github repo with `"registry": "npm"` set in the package.json for easy experimentation, then doing some nasty console.logging down the tree. Happy to look into this, but unfortunately have run out of time for today.
